File: front/src/app/core/interfaces/user-request.ts
/**
 * Represents a request to update or retrieve user information.
 * 
 * @remarks
 * This interface defines the structure of a request used to handle user-related operations,
 * such as updating user details or retrieving specific user information. It includes
 * properties for the user's email, username, and a unique identifier.
 * 
 * @interface
 */
export interface UserRequest {
    
    /**
     * The email address of the user.
     * 
     * @remarks
     * This property holds the user's email address. It is a string value and should be
     * a valid email format.
     * 
     * @example
     * "user@example.com"
     */
    email: string;

    /**
     * The username of the user.
     * 
     * @remarks
     * This property holds the user's username. It is a string value that is unique to
     * the user within the application.
     * 
     * @example
     * "john_doe"
     */
    username: string;

    /**
     * The unique identifier of the user.
     * 
     * @remarks
     * This property is a numeric value that uniquely identifies the user within the
     * system. It is used to reference a specific user record.
     * 
     * @example
     * 1
     */
    id: number;
}

File: back/src/main/java/com/openclassrooms/mddapi/payload/request/UserRequest.java
package com.openclassrooms.mddapi.payload.request;

/**
 * Represents a request payload for updating user information.
 * This class is used to encapsulate the details required to update a user's profile.
 */
public class UserRequest {

    /**
     * The username of the user.
     */
    private String username;

    /**
     * The email address of the user.
     */
    private String email;

    /**
     * The unique ID of the user to be updated.
     */
    private Integer id;

    /**
     * Gets the username of the user.
     * 
     * @return the username
     */
    public String getUsername() {
        return username;
    }

    /**
     * Sets the username of the user.
     * 
     * @param username the username to set
     */
    public void setUsername(String username) {
        this.username = username;
    }

    /**
     * Gets the email address of the user.
     * 
     * @return the email address
     */
    public String getEmail() {
        return email;
    }

    /**
     * Sets the email address of the user.
     * 
     * @param email the email address to set
     */
    public void setEmail(String email) {
        this.email = email;
    }

    /**
     * Gets the unique ID of the user to be updated.
     * 
     * @return the user ID
     */
    public Integer getId() {
        return id;
    }

    /**
     * Sets the unique ID of the user to be updated.
     * 
     * @param id the user ID to set
     */
    public void setId(Integer id) {
        this.id = id;
    }
}
